spec.md

```diff
@@ -4695,17 +4698,40 @@ File = {Statement | newline} eof .
 
 Statement = DefStmt | IfStmt | ForStmt | SimpleStmt .
 
-DefStmt = 'def' identifier '(' [Parameters [',']] ')' ':' Suite .
+DefStmt = 'def' identifier '(' [Parameters [',']] ')' ['->' TypeExpr ]  ':' Suite .
 
 Parameters = Parameter {',' Parameter}.
 
-Parameter  = identifier
-           | identifier '=' Test
+Parameter  = identifier [':' TypeExpr] ['=' Test]
            | '*'
-           | '*' identifier
-           | '**' identifier
+           | '*' identifier [':' TypeExpr]
+           | '**' identifier [':' TypeExpr]
            .
 
+LambdaParameters = LambdaParameter {',' LambdaParameter}.
+
+LambdaParameter  = identifier ['=' Test]
+                 | '*'
+                 | '*' identifier
+                 | '**' identifier
+                 .
+           
+TypeExpr = Type {'|' Type}.
+
+Type = identifier [TypeArguments].
+
+TypeArguments = '[' TypeArgument {',' TypeArgument} ']'.
+
+TypeArgument = TypeExpr
+             | ListOfTypes
+             | DictOfTypes
+             | string
+             .
+
+ListOfTypes = '[' [TypeArgument {',' TypeArgument} [',']] ']'.
+DictOfTypes = '{' [TypeEntry {',' TypeEntry} [',']] '}' .
+TypeEntry    = string ':' TypeArgument .
```

Comment:
Is the key a string or identifier? Is this intended to match a struct field name / record key / dict key?